### PR TITLE
[Hacky] Port from symbolic-shapes - Record TensorReference for ._base tensors, but don't install guards on them  (#89250)

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89735

lint

lint

cc @mlazos @soumith @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/guards.py [hunk 1]
```diff
@@ -580,14 +580,11 @@ def _print_Symbol(self, expr) -> str:
             return "0"
         if expr == 1:
             return "1"
-        expr_found = expr in (self.expr_to_tensor_ref) or (
-            expr in self.intermediary_symbols
+        assert (
+            expr in (self.expr_to_tensor_ref)
+            or (expr in self.intermediary_symbols)
+            or (expr in self.base_symbols)
```
Comment (ezyang) [inline torch/_dynamo/guards.py:588]: you know that the parens here are all unnecessary right

Comment (voznesenskym) [reply torch/_dynamo/guards.py:588]: I find them wonderful for readability, but yes, its all the same to python.

Comment (ezyang) [reply torch/_dynamo/guards.py:588]: `expr in (self.expr_to_tensor_ref)` lol

Comment (voznesenskym) [reply torch/_dynamo/guards.py:588]: fine

**Comment by voznesenskym** [comment]
garbo

**Pull request closed without merging**